### PR TITLE
Updated actions/checkout to v4

### PR DESCRIPTION
**Pull request by ckunki**

### CODE REVIEW

**Review by kaklakariada — APPROVED** [review]

**Review by Nicoretti — CHANGES_REQUESTED** [review]
Sadly, this change is not trivial due to the fact that the upload and download semantics have changed, affecting overwrites and resulting in different download URLs. With v4, if you push the same artifact name multiple times, it gets uploaded multiple times. However, the validity seems to depend on the last "overwrite." If one is unlucky, the wrong download URL will be used on the other end, resulting in broken or non-existing artifacts, etc.

**Comment by ckunki** [comment]
> Sadly, this change is not trivial due to the fact that the upload and download semantics have changed, affecting overwrites and resulting in different download URLs. With v4, if you push the same artifact name multiple times, it gets uploaded multiple times. However, the validity seems to depend on the last "overwrite." If one is unlucky, the wrong download URL will be used on the other end, resulting in broken or non-existing artifacts, etc.

Do you propose to switch back to v3 then?

**Comment by Nicoretti** [comment]
@ckunki we also can push forward this pr, but then we likely need to change the upload and download strategy we are using right now. If this will be addressed in this PR to I'll happy to merge it to.

**Review by kaklakariada — APPROVED** [review]

File: exasol/toolbox/templates/github/workflows/checks.yml [hunk 1]
```diff
@@ -47,21 +47,22 @@ jobs:
...
         with:
           python-version: ${{ matrix.python-version }}
 
-      - name: Run Tests
+      - name: Run lint
         run: poetry run nox -s lint
 
       - name: Upload Artifacts
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
```
Comment (Nicoretti) [inline exasol/toolbox/templates/github/workflows/checks.yml:61]: didn't you intend to revert this?

**Review by Nicoretti — APPROVED** [review]

File: exasol/toolbox/templates/github/workflows/checks.yml (hunk 1 above)
Comment (ckunki) [reply exasol/toolbox/templates/github/workflows/checks.yml]: Thanks.
I fixed your finding.

**Pull request merged**